### ISSUE TITLE
[Enhancement] Some suggestions

### ISSUE THREAD

**Issue by dorpxam** [LongCatFan1/longcat-tuner#32]
## LoRA filename

About the LoRA saved `--save_every_n_epochs`. It's cool. Honestly, I do not use the sampling during training, I prefer save each epoch and multiply the test post-training. Currently, the filename are in format:

- `AAAAAAAA-000000.safetensors`
- `AAAAAAAA-000000_comfy.safetensors`

Where:

- `AAAAAAAA` is the `--output_name` and
- `000000` the epoch number (maybe the step if running with `--save_every_n_steps`)

The idea can be to format the file name with some loss information. It's just informative but can help. For example, a name like:

- `AAAAAAAA-000000-V0.000-A0.000-M0.000-C0.000.safetensors`

Where the additional are respectively:

- `V0.000` loss/video
- `A0.000` loss/audio
- `M0.000` loss/average (M for mean)
- `C0.000` loss/current

And in a ideal world, keeping the system with no change, allowing the format to be set by the user with an option `--output_filename_format`. Something like:

```python
--output_filename_format "{name}-{epoch:08d}-V{video:.3f}-A{audio:.3f}-M{average:.3f}-C{current:.3f}"
```

Where the default value stay:

```python
--output_filename_format "{name}-{epoch:08d}"
```

The `_comfy` is always add to the end of the filename. And here, it can be cool to have a specific output directory for the Comfy LoRA. For example, directly in the dedicated directory. Most simple to manage if you have a hundred of interlaced files (with `_comfy.safetensors` or without `.safetensors`). Like:

```python
--output_comfy_dir "PATH"
```

If not specified, the Comfy LoRA stay in the `--output_dir`.

This option can be cool I think, especially in conjunction with the following proposition.

## Post-Training Inference

This option is more complex but certainly and absolutely necessary I think to allow 'sorting' the different LoRA produced during training.

For allowing multiple testing with the same precise settings. I initially used a JSON object defining the test configuration, which would then be passed to a script—a simpler way to manage and copy configurations. Originally, for the official LTX-2 code (but OOM with), I do something like this:

```json
{
    "paths": {
        "checkpoint_path": "C:/ComfyUI2/models/checkpoints/LTX-2/ltx-2-19b-dev-fp8.safetensors",
        "distilled_lora": "C:/ComfyUI2/models/loras/ltx-2-19b-distilled-lora-384.safetensors",
        "spacial_upscaler": "C:/ComfyUI2/models/latent_upscale_models/ltx-2-spatial-upscaler-x2-1.0.safetensors",
        "gemma_root": "C:/AI-Training/models/LTX-2/gemma-3-12b-it-qat-q4_0-unquantized"
    },
    "config": {
        "seed": 42,
        "width": 1280,
        "height": 704,
        "num_frames": 121,
        "frame_rate": 24.0,
        "num_inference_steps": 40
    },
    "optimization": {
        "fp8transformer": true,
        "distilled": false
    },
    "prompt": {
        "positive": [
            "A female superhero is shooting fireballs out of her hands towards the viewer.",
            "New York City, 1987. At the bottom left of the frame, a girl in a one-piece bathing suit, stockings, and high heels walks along the road, moving away from the camera. Cars drive along the road. The camera quickly zooms in on the girl, visible at the bottom left of the frame. First, we see the girl walking away from us from behind, full-length. Then the camera pans around the girl and reveals a close-up of her delicate, charming face, framed by her lush, curled hair.",
            "A disheveled blonde in a pink dress emerges from the photo booth. Her face is damp. She licks her lips and tries to fix her hair with her hands. Her knees are pink, the same color as her dress, as if she's been standing on them for a long time.",
            "some other prompts ..."
        ],
        "negative": "",
        "enhance": false
    },
    "lora": {
        "path": "E:/LTX-2/ltx2-my-lora/output"
    },
    "output": {
        "format": "{lora_filename}-p{prompt_index:02d}.mp4",
        "path": "E:/LTX-2/ltx2-my-lora/test1"
    }
}
```

Ultimately, the idea is to have precise control over each test by separating the basic configuration from the prompt file (similar to the current `sample_prompts.txt` but in JSON). Something like this:

```json
{
    "prompts": [
        {
            "id": 1,
            "positive": "A disheveled figure emerges from behind a thick billboard, wearing a modern dress with bare shoulders, her face glistening with moisture. She adjusts her corset, straightens her dress, and wipes her face with a handkerchief. A satisfied ebony skinned gentleman emerges from the other side of the billboard and walks in the opposite direction. The disheveled young woman sighs deeply and smiles happily as the gentleman leaves.",
            "negative": "",
            "width": 1280,
            "height": 720,
            "seconds": 10.0
        },
        {
            "id": 2,
            "positive": "A disheveled blonde in a pink dress emerges from the photo booth. Her face is damp. She licks her lips and tries to fix her hair with her hands. Her knees are pink, the same color as her dress, as if she's been standing on them for a long time.",
            "negative": "",
            "width": 1280,
            "height": 720,
            "seconds": 5.0
        },
        {
            "id": 3,
            "positive": "New York City, 1987. At the bottom left of the frame, a girl in a one-piece bathing suit, stockings, and high heels walks along the road, moving away from the camera. Cars drive along the road. The camera quickly zooms in on the girl, visible at the bottom left of the frame. First, we see the girl walking away from us from behind, full-length. Then the camera pans around the girl and reveals a close-up of her delicate, charming face, framed by her lush, curled hair.",
            "negative": "",
            "width": 720,
            "height": 1280,
            "seconds": 5.0
        },
        {
            "id": 13,
            "positive": "A female superhero is shooting fireballs out of her hands towards the viewer.",
            "negative": "",
            "width": 720,
            "height": 1280,
            "seconds": 5.0
        },
        {
            "id": 4,
            "positive": "A woman walks along the beach by the sea. The waves are raging, and this woman is dancing, smiling, and enjoying life. She is dressed lightly, in a summery way. She has blonde hair and blue eyes. She is incredibly beautiful and young.",
            "negative": "",
            "width": 1024,
            "height": 1024,
            "seconds": 5.0
        }
    ]
}
```
Where the parameters (`width / height / negative` ...) override the parameters defined in the configuration file (`config.json`), which points to the prompts file (`prompts.json`). This allows for easy and methodical testing of different prompt sets.

This is far from a negligible option for testing LoRa produced during training. Similarly, generating an image of the thumbnails of multiple LoRa within a single image (by selecting the first frame, or even a specific frame in the video) would allow for an at-a-glance view of the differences between the various stages of the training progression.

This is a fairly straightforward option; here, there's no need for full resolution, just an overview. For example, if we have 32 trained LoRa, an 8-row array of 4 thumbnails saved in an image file (corresponding to the same prompt, of course) would provide a preview before analyzing each video file for more detail.

I could code the batch processing code; in fact, I tried using the current code to create a tool, but the current inference process is heavily intertwined with the training process, and I'm afraid of making mistakes.

## Conclusion

What do you think? There's no rush, obviously, but I believe these are good ideas for thoroughly testing what the training code produces. Ideally, by preserving the training configuration information and properly testing the LoRa, we can refine the parameters to clearly visualize the implications of the different training options.

As always, thank you for everything.

**Comment by AkaneTendo25** [comment]
Thank you for the suggestions. I think they're solid at their core, but I'd start with creating a metadata file alongside each checkpoint with all relevant info (loss, learning rate, etc.) Saving it could be controlled by a flag (e.g. `--save_metadata`). This way it's easy to analyze checkpoints with any script and not bind to a specific naming format. Some of this data already exists when `--save_state` is enabled, but having a small metadata file per LoRA could still be convenient. Also, there's some metadata embedded into each checkpoint that can be inspected by analyzing the LoRA file itself.

I also thought about exposing flags to control the output format (save native LoRA, ComfyUI-only, or both).

Anyway, since these changes don't touch the training core, they should be pretty easy to implement, so I'll think about integrating them.

The post-training inference and comparison tooling you describe makes a lot of sense, but I'd prefer to tackle that separately once inference is a bit more decoupled from training.

**Comment by dorpxam** [comment]
> The post-training inference and comparison tooling you describe makes a lot of sense, but I'd prefer to tackle that separately once inference is a bit more decoupled from training.

Of course. I'm glad you're on board with these ideas. Really, really cool!

**Comment by dorpxam** [comment]
The latest change save only 'ComfyUI' LoRAs by default. It's cool, but to be sure that the change is clear for the user, a (temporary) colored warning in the console before starting the training can be cool. I say that because the non-ComfyUI LoRA stay useful for restart the training. 

I don't know if the Comfy-LoRA can be converted back to non-Comfy LoRA because if I'm not mistaken, some layers are ignored, aren't they?

**Comment by AkaneTendo25** [comment]
> I say that because the non-ComfyUI LoRA stay useful for restart the training.

I completely forgot about that! You’re right. It's better to make this configurable while still producing both LoRAs by default.

(It's possible to convert it back to the original format on the fly, but that would introduce unnecessary complexity. If user really wants to get rid of non-comfy checkpoints, they can control it with an explicit flag.)

**Comment by dorpxam** [comment]
> I completely forgot about that!

Yes. If the default stay both non-comfy/comfy. Perhaps a new filename strategy can be set by default to avoid interlacing files (faster copy/paste of the files).

Current filename is:
```
AAAA-000000.safetensors
AAAA-000000_comfy.safetensors
```
Maybe:
```
AAAA-000000.safetensors
AAAA-comfy-000000.safetensors
```
Can allow to have separated files.

OR double extension:
```
AAAA-000000.comfy.safetensors
```
OR prefix:
```
Comfy-AAAA-000000.safetensors
```
In all case, avoid:
```
000000_comfy
```

Thanks a lot!
